pilot/pkg/model/authentication.go

```diff
@@ -160,3 +161,104 @@ func getConfigsForWorkload(configsByNamespace map[string][]Config,
 
 	return configs
 }
+
+
+// getMutualTLSMode returns the MutualTLSMode enum corresponding peer MutualTLS settings.
+// Input cannot be nil.
+func getMutualTLSMode(mtls *v1beta1.PeerAuthentication_MutualTLS) MutualTLSMode {
```

Comment:
Can we not move this code here? I'd like to keep the details how the API is used separate from infra to serve them (i.e this package)

Reply:
i tried, but failed. because otherwise we have circular dependency.

model depends on v1bate1 for policy resolution, and v1beta1 depends on model again. after consideration, this seems most cost-effective refactoring i can think of.

Reply:
i might be able to define interface in model package and impl that in authn one. i'll give it a try.